src/lib.rs

```diff
@@ -185,6 +185,23 @@ impl EventHandler for std::sync::mpsc::Sender<Result<Event>> {
     }
 }
 
+/// Watcher kind enumeration
+#[derive(Debug,PartialEq,Eq)]
+pub enum WatcherKind {
```

Comment:
I guess we'll add/remove watchers in the future (actually, kqueue has been added recently) so it should make sense to add the `#[non_exhaustive]` attribute.